Client/Frontend/Browser/Tab Tray/TabTrayController.swift

```diff
@@ -146,6 +149,16 @@ class TabTrayController: LoadingViewController {
       initialScrollCompleted = true
     }
   }
+  
+  override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+    super.traitCollectionDidChange(previousTraitCollection)
+
+    searchBarView?.frame = navigationController?.navigationBar.frame ?? .zero
```

Comment:
Maybe do not call this code if horizontal size class did not change? add something like 
```
if previousTraitCollection?.horizontalSizeClass == traitCollection.horizontalSizeClass { return }
```

need to be checked if this works

Reply:
This doesn't not since the main part that makes this work is fetching the frame of navigation bar every time trait change happens and setting it to search bar View and updating titleView. This is **not** a costly operation but has to be done to make titleView work properly.